src/components/container/MyDataHelpsAssistant/MyDataHelpsAssistant.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faPaperPlane } from '@fortawesome/free-solid-svg-icons/faPaperPlane';
+import { faUser } from '@fortawesome/free-solid-svg-icons/faUser';
+import { faGear } from '@fortawesome/free-solid-svg-icons/faGear';
+import { faFlask } from '@fortawesome/free-solid-svg-icons/faFlask';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons/faSpinner';
+import { faChevronDown } from '@fortawesome/free-solid-svg-icons/faChevronDown';
+import { StreamEvent } from '@langchain/core/tracers/log_stream';
+import { AIMessageChunk } from '@langchain/core/messages';
+import { StructuredTool } from '@langchain/core/tools';
+import Markdown from 'react-markdown';
+import { MyDataHelpsAssistant } from '../../../helpers/assistant/assistant';
+
+import '@fortawesome/fontawesome-svg-core/styles.css';
+import "./MyDataHelpsAssistant.css";
+
+export interface MyDataHelpsAssistantProps {
+    innerRef?: React.Ref<HTMLDivElement>;
+    previewState?: "default";
+    debug: boolean;
+    additionalInstructions?: string;
+    tools?: StructuredTool[];
+    appendTools?: boolean;
+}
+
+export interface MyDataHelpsAssistantMessage {
+    type: string;
+    content: string;
+    runId?: string;
+}
+
+export default function (props: MyDataHelpsAssistantProps) {
```

Comment:
As I look at this I think it'd be ideal to extract a generic generic presentational "mdhui-chat" component; such that the AI bits aren't tied up with the assistant bits.

So here's what I would do:
- Create a "Chat" component under presentational
- As props it would have:
  - messages; each message would have an icon
  - an "onMessageSent" function which captures the user's next message they send

Then the "AIAssistant" container component would just use this "Chat" component to render the various messages it's passing back and forth etc...and could be used for other use cases later